src/main/java/billgates/entities/UserFactory.java

```diff
@@ -0,0 +1,30 @@
+package billgates.entities;
+
+public class UserFactory {
+    //use getShape method to get object of type shape
+    public IUser getUser(int id, String name, String UserType) {
+        if (UserType == "DummyUser") {
+            IUser d = new DummyUser(id, name);
+            return d;
+        }
+        if (UserType == "User") {
+            IUser user = User.getInstance(id, name, "TemporyPW", 0);
+
+            return user;
+        }
+        return null;
+    }
+
+    public IUser getUser(int id, String name,String password,int billID,String UserType) {
```

Comment:
Hmm, this looks pretty much like a "Code smell" that we are creating redundant lines by copy pasting from the above, maybe my suggestion could adhere to the factory method better.